src/test/CMakeLists.txt

```diff
@@ -126,6 +126,28 @@ macro(add_shadow_tests)
          set(POST_CMD "true")
       endif()
 
+      if(SHADOW_TEST_CHECK_RETVAL)
+         # Validate that all processes exited with code 0.  Processes that don't
+         # exit have an empty exitcode file, which will cause this check to
+         # fail.
+         # We *don't* fail in the case that nothing matched the glob, which is
+         # why we need to explicitly check whether the file exists.
+         set(CHECK_RETVAL_CMD "
+             for f in ${SHADOW_TEST_NAME}.data/hosts/*/*.exitcode
+             do
+               # Check if glob actually matched
+               if [ ! -f $f ]\; then continue\; fi
```

Comment:
I think you can just set `shopt -s nullglob` before the `for`, and you don't need to worry about this condition.

Reply:
`nullglob` is bash-specific, and it looks like CMake is using /usr/bin/sh. Maybe there's a way to force it to use bash, but I don't think it's worth forcing the dependency